src/Parser.jl

```diff
@@ -421,20 +442,22 @@ end
 
 function parse(io::IO;
                dicttype=Dict{String,Any},
-               inttype::Type{<:Real}=Int64)
-    pc = _get_parsercontext(dicttype, inttype)
+               inttype::Type{<:Real}=Int64,
+               allownan=false)
+    pc = _get_parsercontext(dicttype, inttype, allownan)
     ps = StreamingParserState(io)
     parse_value(pc, ps)
 end
 
 function parsefile(filename::AbstractString;
                    dicttype=Dict{String, Any},
                    inttype::Type{<:Real}=Int64,
+                   allownan=false,
```

Comment:
Maybe we should default to `allownan=true`, i.e. permissive by default?

Reply:
I was actually going to suggest this as well.  There's precedent--e.g., the Google GSON library does this, even though it doesn't produce these values by default (and neither do we).

Reply:
The last commit sets `allownan=true` by default.  I think this is a good change, but want to make sure there's consensus before merging.